internal/plans/objchange/objchange.go

```diff
@@ -286,64 +286,58 @@ func proposedNewAttributes(attrs map[string]*configschema.Attribute, prior, conf
 		}
 
 		configV := config.GetAttr(name)
+
 		var newV cty.Value
 		switch {
-		case attr.Computed && attr.Optional:
-			// This is the trickiest scenario: we want to keep the prior value
-			// if the config isn't overriding it. Note that due to some
-			// ambiguity here, setting an optional+computed attribute from
-			// config and then later switching the config to null in a
-			// subsequent change causes the initial config value to be "sticky"
-			// unless the provider specifically overrides it during its own
-			// plan customization step.
-			if configV.IsNull() {
-				newV = priorV
-			} else {
-				newV = configV
-			}
-		case attr.Computed:
+		// required isn't considered when constructing the plan, so attributes
+		// are essentially either computed or not computed. In the case of
+		// optional+computed, they are only computed when there is no
+		// configuration.
+		case attr.Computed && configV.IsNull():
 			// configV will always be null in this case, by definition.
 			// priorV may also be null, but that's okay.
 			newV = priorV
+		case attr.NestedType != nil:
+			// For non-computed NestedType attributes, we need to descend
```

Comment:
It's effectively non-computed because the config must be non-null, which is check in the case above. If it were optional+computed being treated as computed it's still caught in the first case.